File: Practica 1/Punto.py
import math

class Punto:
    def __init__(self, x: float, y: float):
        self.x = x
        self.y = y

    def coord_cartesianas(self):
        return f"Coordenadas Cartesianas: ({self.x}, {self.y})"

    def coord_polares(self):
        r = math.sqrt(self.x**2 + self.y**2)
        theta = math.atan2(self.y, self.x)
        return f"Coordenadas Polares: (r={r:.2f}, θ={math.degrees(theta):.2f}°)"

    def __str__(self):
        return f"Punto({self.x}, {self.y})"

####
p = Punto(3.0, 4.0)
print(p)
print(p.coord_cartesianas())
print(p.coord_polares())

File: Practica 1/Punto.java

public class Punto {
	public float x;
	public float y;
	
	public Punto(float x, float y) {
		this.x = x;
		this.y = y;
	}
	public String coord_cartesianas() {
		return "(" + x + ", " + y +")";
	}
	public String coord_polares() {
		double r = Math.sqrt(x * x + y * y);
        double theta = Math.toDegrees(Math.atan2(y, x));
        return String.format("Coordenadas Polares: (r=%.2f, θ=%.2f°)", r, theta);
	}
	public String toString() {
		return "Punto(" + x + ", " + y + ")";
	}
	public static void main(String[] args) {
		Punto p = new Punto(3.0f, 4.0f);
        System.out.println(p);
        System.out.println(p.coord_cartesianas());
        System.out.println(p.coord_polares());
	}

}
